browser/src/control/Control.JSDialog.js

```diff
@@ -401,6 +405,35 @@ L.Control.JSDialog = L.Control.extend({
 		}
 	},
 
+	 findTabControl: function(obj) {
+		if (obj.type === 'tabcontrol') {
+			return obj;
+		}
+		if (obj.children && obj.children.length > 0) {
+			for (let child of obj.children) {
+				const result = this.findTabControl(child); // Recursively search in children
+				if (result) return result; // If found, return the tabcontrol
+			}
+		}
+		return null; // Return null if tabcontrol is not found
+	},
+
+	 focusOnTabControl: function(dialogObject) {
+
+		const tabControl = this.findTabControl(dialogObject);
+	
+		if (tabControl && tabControl.tabs && tabControl.tabs.length > 0) {
+			// Construct the id using 'type' + tabs[0].id
+			const targetId = tabControl.id + '-' + tabControl.tabs[0].id;
```

Comment:
This is is generated as same method we do while creating any tabcontrol inside a tab dialog

Reply:
This is fragile - if we will change that, eg. due to some a11y rework, we will lost this functionality.
So if we would want this approach it should be done using shared function in both cases.
This is why I don't want using hardcoded ids, it can be easily broken. Also in jsdialogs we sometimes rename elements using some suffix with number if we have duplicated nodes (what can happen in some specific situations - and we don't test all possibledialogs with any other one :) )

Also - I would prefere other method: use helper which takes container and returns first focusable element:
this one https://github.com/CollaboraOnline/online/blob/master/browser/src/control/jsdialog/Util.FocusCycle.js#L117

This makes our code very unified. We find for focusable elements in the same way everywhere.